jakartaee9/javahostel/src/main/resources/META-INF/persistence.xml

```diff
@@ -1,10 +1,29 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<persistence version="2.1" xmlns="http://xmlns.jcp.org/xml/ns/persistence" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/persistence http://xmlns.jcp.org/xml/ns/persistence/persistence_2_1.xsd">
-	<persistence-unit name="JavaHostel">
+<persistence xmlns="https://jakarta.ee/xml/ns/persistence" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="https://jakarta.ee/xml/ns/persistence https://jakarta.ee/xml/ns/persistence/persistence_3_0.xsd" version="3.0">
+	<persistence-unit name="JavaHostel" transaction-type="JTA">
 		<provider>org.hibernate.ejb.HibernatePersistence</provider>
-		<jta-data-source>java:jboss/datasources/javahostel</jta-data-source>
+		<jta-data-source>java:app/datasources/javahostel</jta-data-source>
+
 		<properties>
-			<property name="hibernate.hbm2ddl.auto" value="update"/>
+			<property name="javax.persistence.schema-generation.database.action" value="create" />
```

Comment:
Essa troca mantém o fato de que se uma classe é modificada a tabela é atualizada no próximo deploy? Como o valor muda de update para create, sempre achei que esta funcionalidade fosse perdida...

Reply:
90% de certeza que sim, mas vou confirmar.

Reply:
Ok, executei alguns testes e realmente não funciona igual. De acordo com a espeficicação, para a propriedade 'jakarta.persistence.schema-generation.database.action' são 4 valores possíveis: none, create, drop-and-create, drop.

Com o valor em 'create', aparentemente o provider vai sempre tentar executar a criação das tabelas de acordo com o mapeamento jpa. Então, caso exista uma entidade nova, vai funcionar (apesar do deploy ficar bem feio pois gera um monte de warning para as tabelas que já existem).

Por outro lado, caso você adicione uma propriedade a uma entidade já existente, não vai funcionar (pois ele não gera um alter table). Para esse caso, o 'hibernate.hbm2ddl.auto' em 'auto' realmente funciona. Mas também tem suas limitações. Eu tentei remover essa propriedade da classe e a coluna continuou existindo na tabela.

Mas se warning não for um problema, sempre é possível usar o drop-and-create também. A primeira vez que for executado o deploy, vai gerar alguns warnings porque as tabelas não existirão.

Enfim, foi até bom revisar isso porque eu não tinha me atentado que ainda estava usando os nomes das propriedades como javax.